multicloud/src/main/java/de/uniulm/omi/cloudiator/sword/multicloud/pricing/aws/AWSPricingSupplier.java

```diff
@@ -0,0 +1,204 @@
+                                Map.Entry<String, BigDecimal> price = entry3.getValue().getPricePerUnit().entrySet().iterator().next();
+                                pricingDimensions.add(new PricingDimensions(price.getValue(), entry3.getValue().getUnit(), entry3.getValue().getDescription(), entry3.getValue().getBeginRange(), entry3.getValue().getEndRange()));
+                            }
+                            pricingReservedTerms.add(new PricingTerms(pricingDimensions, entry.getValue().termAttributes.get("LeaseContractLength"), entry.getValue().termAttributes.get("OfferingClass"), entry.getValue().termAttributes.get("PurchaseOption")));
+                        }
+                    }
+                    if(!currency.isPresent()) {
+                        currency = Optional.ofNullable(awsProductAndTerms.terms.get("Reserved").entrySet().iterator().next().getValue().getAwsDimensionRates().entrySet().iterator().next().getValue().entrySet().iterator().next().getValue().getPricePerUnit().entrySet().iterator().next().getKey());
+                    }
+                }*/
+
+                Pricing pricing = PricingBuilder.newBuilder()
+                        .id(awsProductAndTerms.product.sku)
+                        .providerId(awsProductAndTerms.product.sku)
+                        .name(awsProductAndTerms.product.attributes.get("usagetype"))
+                        .location(null)
+                        .locationProviderId(awsPricingLocationConverter.apply(awsProductAndTerms.product.attributes.get("location")))
+                        .instanceName(awsProductAndTerms.product.attributes.get("instanceType"))
+                        .cloudServiceProviderName("aws")
+                        .api(ApiBuilder.newBuilder().providerName("aws-ec2").build())
+                        .currency(currency.isPresent() ? currency.get() : "N/A")
+                        .tenancy(awsProductAndTerms.product.attributes.get("tenancy"))
+                        .operatingSystem(operatingSystemBuilder.build())
+                        .onDemandTerms(pricingOnDemandTerms)
+                        .reservedTerms(pricingReservedTerms)
+                        .licenseModel(awsProductAndTerms.product.attributes.get("licenseModel"))
+                        .capacityStatus(awsProductAndTerms.product.attributes.get("capacitystatus"))
+                        .preInstalledSw(awsProductAndTerms.product.attributes.get("preInstalledSw"))
+                        .productFamily(awsProductAndTerms.product.productFamily)
+                        .operation(awsProductAndTerms.product.attributes.get("operation"))
+                        .build();
+                pricings.add(pricing);
+            }
+
+            nextToken = awsPricingPack.getNextToken();
+        } while (nextToken != null);
+        LOGGER.info(String.format("AWS products (with terms) processed (= size of returned Set<Pricing>): %d", pricings.size()));
```

Comment:
Logger has support for placeholders, so its the same:
LOGGER.info("AWS products (with terms) processed (= size of returned Set<Pricing>): {}}", pricings.size());

Reply:
done